### ISSUE TITLE
Data inconsistency in non clustered PK after `load data ...replace into ...`

### ISSUE THREAD

**Issue by fzzf678** [pingcap/tidb#56408]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
➜  ~ cat /Users/fanzhou/a.csv
1|aa|beijing
1|aa|beijing
1|aa|beijing
1|aa|beijing
2|bb|shanghai
2|bb|shanghai
2|bb|shanghai
3|cc|guangzhou
```

```
mysql> create table a(id int, name varchar(20), addr varchar(100), primary key (id) nonclustered);
Query OK, 0 rows affected (0.02 sec)

mysql> load data local infile '/Users/fanzhou/a.csv' replace into table a fields terminated by '|' escaped by '' lines terminated by '\n';
Query OK, 13 rows affected (0.01 sec)
Records: 8  Deleted: 0  Skipped: 0  Warnings: 0

mysql> select count(*) from a;
+----------+
| count(*) |
+----------+
|        3 |
+----------+
1 row in set (0.01 sec)

mysql> select count(*) from a use index();
+----------+
| count(*) |
+----------+
|        8 |
+----------+
1 row in set (0.01 sec)

mysql> admin check table a;
ERROR 8223 (HY000): data inconsistency in table: a, index: PRIMARY, handle: 3, index-values:"" != record-values:"handle: 3, values: [KindInt64 1]"
```

### 2. What did you expect to see? (Required)
Data consistent, No error
### 3. What did you see instead (Required)
Data Inconsistency
### 4. What is your TiDB version? (Required)
master

**Comment by wjhuang2016** [comment]
``` sql
mysql> show create table te4e81423;
+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table     | Create Table                                                                                                                                                 |
+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
| te4e81423 | CREATE TABLE `te4e81423` (
  `col_21` decimal(26,14) NOT NULL,
  UNIQUE KEY `idx_15` (`col_21`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci |
+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

csv:
```
0
834382.8322
3290.508
3818.726
409.797
3901
97
95.308
11.71
9.437
355.51
3364
8.11
859.735
```

**Comment by wuhuizuo** [comment]
/remove-label affects-8.5

it's found and fixed before v8.5.0 release.
